src/xlsx.js

```diff
@@ -21,27 +21,28 @@ export class Workbook {
   }
 }
 
-function extract(sheet, {range, headers = false} = {}) {
+function extract(sheet, {range, headers} = {}) {
   let [[c0, r0], [c1, r1]] = parseRange(range, sheet);
-  const headerRow = headers && sheet._rows[r0++];
+  const headerRow = headers ? sheet._rows[r0++] : null;
   let names = new Set(["#"]);
   for (let n = c0; n <= c1; n++) {
-    let name = (headerRow ? valueOf(headerRow._cells[n]) : null) || toColumn(n);
+    let name = (headerRow ? valueOf(headerRow._cells[n]) : null) || toColumn(n); // should a falsey header value be ignored? or just nullish?
     while (names.has(name)) name += "_";
     names.add(name);
   }
   names = new Array(c0).concat(Array.from(names));
 
   const output = new Array(r1 - r0 + 1);
   for (let r = r0; r <= r1; r++) {
-    const row = (output[r - r0] = Object.defineProperty({}, "#", {
+    // Should we be using Object.create(null) instead of an empty object here?
+    const row = (output[r - r0] = Object.defineProperty({}, "#", { // what is this non-enumerable row["#"] property for?
       value: r + 1,
     }));
-    const _row = sheet._rows[r];
+    const _row = sheet._rows[r]; // is this an internal ExcelJS API? why not sheet.getRow(r)?
```

Comment:
Optimization? Can change to `getRow`, which means a row is always returned (if it was previously missing getRow creates one) and the check for `_row &&` below would be unneeded.

Reply:
Oof, for tests I want to bring in exceljs as a devDependency then. The minimal mockWorkbook function won't cut it anymore.

Reply:
I did this in #255 as another PR on top of this one as it brings in ExcelJS as a devDependency to use in tests.